evals/elsuite/utils.py

```diff
@@ -135,3 +144,31 @@ def __call__(self, **kwargs):
             **self.completion_kwargs,
         )
         return completion, prompt
+
+
+class CompletionFn(Protocol):
+
+    def __call__(
+        self,
+        model_spec: ModelSpec,
+        prompt: Union[OpenAICreatePrompt, OpenAICreateChatPrompt, Prompt],
+        **kwargs
+    ) ->tuple[dict, Union[OpenAICreatePrompt, OpenAICreateChatPrompt], dict]:
```

Comment:
To provide some context around the `completion_query` return signature (which this follows):

- `actual_prompt` (2nd item) is only different from `prompt` when we use `strip_completion`, which isn't applicable here
- `metadata` (3rd item) is derived from `result` (1st item), so it's actually not necessary

---

Currently this return signature only applies to OpenAI models. Are we planning to add support for using non-OpenAI models here? If so, we should use a more generic return signature. One option could be

```
-> CompletionResult

@dataclass
class CompletionResult:
  completions: List[str]
  oai_completion_response: Optional[dict] = None
  # add more as needed for non-OpenAI models?
```